File: app/src/main/java/org/michaelbel/moviemade/ui/base/BaseFragment.java
package org.michaelbel.moviemade.ui.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.michaelbel.moviemade.moxy.MvpAppCompatFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseFragment extends MvpAppCompatFragment {

    private View parentView;
    private Unbinder unbinder;

    protected abstract int getLayout();

    protected View getParentView() {
        return parentView;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        parentView = inflater.inflate(getLayout(), container, false);
        unbinder = ButterKnife.bind(this, parentView);
        return parentView;
    }

    // TODO replace with it. Remove getLayout. Add onCreateView in each fragment.
    /*@Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        unbinder = ButterKnife.bind(this, view);
    }*/

    @Override
    public void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/base/BaseFragment.kt
package org.michaelbel.moviemade.presentation.base

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import butterknife.ButterKnife
import butterknife.Unbinder

abstract class BaseFragment: Fragment() {

    private var unbinder: Unbinder? = null

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        unbinder = ButterKnife.bind(this, view)
    }

    override fun onDestroy() {
        super.onDestroy()
        unbinder!!.unbind()
    }
}